docs: document where to find changelog for base snaps (#6314)
snapcore/snapcraft: docs/how-to/change-bases/index.rst

.. meta::

>     :description: How to change the base snap of a snap and access its ChangeLog.

Reviewer: Rewriting this one because I have domain knowledge of it:

```suggestion
    :description: How to change the base of a snap. Switch between bases like core24 to core26. This guide describes the different requirements of each migration path.
```

Author: done

    change-from-core20-to-core22
    change-from-core22-to-core24
    change-from-core24-to-core26

Base changelogs
---------------

Each base snap contains a changelog at ``/usr/share/doc/ChangeLog``
that summarizes the package updates bundled in that release,
along with the aggregated updates from previous releases.
In the examples below, replace ``<base>`` with the name of
your base snap (for example, core24).

When the snap is already installed
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

If the base snap is already installed on the system, the changelog is available
at ``/snap/<base>/current/usr/share/doc/ChangeLog``.

You can list all installed revisions and versions with ``snap list --all <base>``.

> To inspect a specific revision that is not the active one, replace ``current`` with

Reviewer: ```suggestion
You can list all installed revisions and versions of a base with ``snap list --all <base>``.
To inspect a specific revision, replace ``current`` with
```

Author: done

    change-from-core20-to-core22
    change-from-core22-to-core24
    change-from-core24-to-core26

> Changelog for base snaps

Author: done

> ChangeLog for base snaps

Reviewer: Please update all instances of "ChangeLog":

```suggestion
Base changelogs
```

Author: done. Of course, the references to the actual file named `ChangeLog` is still kept

-------------------------

> Each release of the base snap includes a ChangeLog at

Reviewer: ```suggestion
Each base snap contains a ChangeLog at
```

Author: done

> ``/usr/share/doc/ChangeLog`` that summarises the package updates bundled in

Reviewer: ```suggestion
``/usr/share/doc/ChangeLog`` that summarizes the package updates bundled in
```

We switched to US English last year.

Author: done

> ``/usr/share/doc/ChangeLog`` that summarises the package updates bundled in

Reviewer: When I looked at core24's changelog, it seemed like all the changes were from us, and not upstream package development. Does the changelog include everything that went into the package's history, or only what Canonical patched?

Author: it includes whatever changelog content has been recorded while releasing the package in Ubuntu, which may or may not contain some notes of upstream changes. As an example https://changelogs.ubuntu.com/changelogs/binary/libd/libdrm2/2.4.125-1/changelog contains some content from upstream changes.
Let me know if you would like me to reword this.

that release, along with the aggregated updates from previous releases.
In the examples below, replace ``<base>`` with the name of
your base snap (for example, ``core24``). Note that ``core26`` does not

> currently include a ChangeLog.

Reviewer: ```suggestion
your base snap (for example, core24).

Core26 doesn't currently include a changelog.
```

We only put core names in code formatting when we're specifically referring them in `snapcraft.yaml`.

Author: done

When the snap is already installed
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

If the base snap is already installed on the system, the ChangeLog is available
at ``/snap/<base>/current/usr/share/doc/ChangeLog``.

> You can find all the available revisions with ``snap list --all <base>``. To inspect

Author: done

> a specific revision that is not the active one, replace ``current`` with the

Reviewer: Can we give the user guidance about how to check which revision corresponds to which core patch version?

Author: i've added that `list all installed revisions and versions...` as the version can be seen together with revision in the output of `snap list`.

> revision number (e.g. ``/snap/<base>/<revision>/usr/share/doc/ChangeLog``).

Reviewer: ```suggestion
revision number. For example, ``/snap/<base>/<revision>/usr/share/doc/ChangeLog``.
```

Author: done

Via ``snap download``

> ~~~~~~~~~~~~~~~~~~~~~

Reviewer: ```suggestion
Download an unused base
~~~~~~~~~~~~~~~~~~~~~~~
```

Author: `unused` doesn't seem good here, i've instead made it `Download a base`

You can download a snap without installing it, then extract the

> ChangeLog from the squashfs image:

Reviewer: ```suggestion
You can download a base without installing it, then extract the
ChangeLog from it:
```

Author: done

.. code-block:: bash

    # Download the snap
    snap download <base>

>     # Extract the changelog into a local directory

Author: done

>     unsquashfs -d <base>-unpacked <base>_*.snap usr/share/doc/ChangeLog

Reviewer: ```suggestion
    snap download <base>
    unsquashfs -d <base>-unpacked <base>_*.snap usr/share/doc/ChangeLog
```

Author: done
